### PR TITLE
Fix bbox prediction for torch.Tensor source

### LINKED ISSUE #1371: [Detection] Results misbehave with torch.tensor as input
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### YOLOv8 Component

Detector

### Bug

Reproduce: Call ultralytics.yolo.engine.model.YOLO.predict with input source as a batch of images in the form of Tensor[B, C, H, W]. Please make sure that the batch images contains as least one valid object so that the model can output a detection.

Expected behavior: The call returns a B long list containing Results objects. Each Results object have attribute orig_shape=[H, W], and attribute boxes containing the detection bounding boxes.

Observed behavior: The call returns a B long list containing Results objects. Each Results object have attribute orig_shape=[B, C]. These Results object have attribute boxes containing the detection bounding boxes, which has the correct classes and confidence values, but coordinates (xyxy, xyhw,...) all based on the incorrect assumption that original image shape is [B,C].

### Environment

ultralytics==8.0.52
torch==1.11.0+cu113

### Minimal Reproducible Example

_No response_

### Additional

_No response_

### Are you willing to submit a PR?

- [x] Yes I'd like to help by submitting a PR!

### PR DESCRIPTION
**Pull request by hotfur**
#1371 

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://github.com/ultralytics/actions)<sub>

### 🌟 Summary
Enhanced image postprocessing compatibility for object detection and segmentation in YOLOv8.

### 📊 Key Changes
- Modified scaling of bounding boxes to only take effect when `orig_imgs` is not a `torch.Tensor`.
- Eliminated redundant `shape` variable and directly referenced `orig_img.shape` when scaling bounding boxes.
- Ensured consistent handling of scaling across both detection and segmentation functions.

### 🎯 Purpose & Impact
- **Purpose:** To streamline postprocessing when handling predictions, enabling the system to be more adaptable to different types of image inputs.
- **Impact:** Improves flexibility in object detection and segmentation tasks, potentially increasing accuracy and efficiency when `orig_imgs` are not tensors. This change is non-disruptive to existing workflows but allows for more robust handling of diverse input formats. 🏗️📈

### CODE REVIEW

**Comment by hotfur** [comment]
recheck

**Comment by hotfur** [comment]
I have read the CLA Document and I sign the CLA

**Comment by hotfur** [comment]
recheck

**Comment by glenn-jocher** [comment]
@AyushExel @Laughing-q not sure we're rounding prediction boxes here, I'm going to remove this part.

**Comment by glenn-jocher** [comment]
@hotfur PR is merged. Thank you for your contributions to YOLOv8 🚀 and Vision AI ⭐
